tests_and_analysis/test/euphonic_test/test_castep_reader.py

```diff
@@ -41,12 +41,17 @@ def check_dict(dct, expected_dct):
                     check_dict(dct[exp_key], exp_val)
                 else:
                     assert dct[exp_key] == exp_val
-        # Temporary solution until test data has been regenerated
-        # units have changed
-        expected_dos_data['dos_unit'] = 'meV'
-        dos_conv = (1*ureg('1/cm').to('meV')).magnitude
+
+        # Convert ref data from cm to 1/meV to match current behaviour
+
+        expected_dos_data['dos_unit'] = '1/meV'
+
         for key, value in expected_dos_data['dos'].items():
-            expected_dos_data['dos'][key] = [x/dos_conv for x in value]
+            expected_dos_data['dos'][key] = (ureg.Quantity(value, "cm")
+                                             .to("1 / meV", "reciprocal_spectroscopy")
+                                             .magnitude
+                                             .tolist())
```

Comment:
I mean, this is fine, but should it be a dict comprehension?

Reply:
It could be, but
- performance difference is irrelevant, this is a test
- readability is already an issue here, dict comprehension would make it worse, I think?

```python
        expected_dos_data['dos'] = {
            key: ureg.Quantity(value, "cm")
                    .to("1 / meV", "reciprocal_spectroscopy")
                    .magnitude
                    .tolist()
            for key, value in expected_dos_data["dos"].items()
        }

```

I think the nicest option is 

```python
        from toolz.dicttoolz import valmap

        def transform(dos: list[float]) -> list[float]:
            return (ureg.Quantity(dos, "cm")
                    .to("1 / meV", "reciprocal_spectroscopy")
                    .magnitude
                    .tolist())

        expected_dos_data["dos"] = valmap(transform, expected_dos_data["dos"])
```

but toolz is not yet a Euphonic dependency. (I'd like to start using it for some other stuff, and Mantid team seems to be open to this. I suppose at this point it would just be a testing dependency.)

Without toolz there is

```python
        def transform(dos: list[float]) -> list[float]:
            return (ureg.Quantity(dos, "cm")
                    .to("1 / meV", "reciprocal_spectroscopy")
                    .magnitude
                    .tolist())

        expected_dos_data["dos"] = {
            key: transform(value)
            for key, value in expected_dos_data["dos"].items()}
```

or

```python
        def transform(dos_item: tuple[str, list[float]]) -> list[float]:
            key, dos = dos_item
            return key, (ureg.Quantity(dos, "cm")
                         .to("1 / meV", "reciprocal_spectroscopy")
                         .magnitude
                         .tolist())

        expected_dos_data["dos"] = dict(
            map(transform, expected_dos_data["dos"].items()))
```

Reply:
Yeah, doesn't make much difference. Wish `toolz` become somewhat core like `itertools` and things. At least dict key/value maps.

Reply:
Well, as I do ultimately intend to use it in the main Euphonic code, perhaps it's not a bad idea to add it to the test requirements now and see where else it can sneak into use.